numpy/polynomial/_polybase.py

```diff
@@ -67,6 +67,42 @@ class ABCPolyBase(abc.ABC):
     # Limit runaway size. T_n^m has degree n*m
     maxpower = 100
 
+    # Unicode character mappings for improved __str__
+    _superscript_mapping = {
+        0 : "\u2070",
+        1 : "\u00B9",
+        2 : "\u00B2",
+        3 : "\u00B3",
+        4 : "\u2074",
+        5 : "\u2075",
+        6 : "\u2076",
+        7 : "\u2077",
+        8 : "\u2078",
+        9 : "\u2079"
```

Comment:
Make the keys strings, and then you can use `str.translate` below

Reply:
That's a great idea - better to use built-in functionality than reinvent the wheel. Addressed in 380c9f0d3